### PR TITLE
fix: legacy install of npm and yarn

### PR DESCRIPTION
**Pull request by Chumper**
Should fix legacy install of npm and yarn

### CODE REVIEW

**Comment by viceice** [comment]
"/tmp/empty-cache":  looks like we need to create this with 775 too

File: src/usr/local/buildpack/tools/yarn.sh [hunk 1]
```diff
@@ -7,11 +7,7 @@ check_command node
...
 
 if [[ -z "${tool_path}" ]]; then
-  INSTALL_DIR=$(get_install_dir)
-  base_path=${INSTALL_DIR}/${TOOL_NAME}
-  tool_path=${base_path}/${TOOL_VERSION}
-
-  mkdir -p "${tool_path}"
+  tool_path="$(create_versioned_tool_path)"
 
   NPM_CONFIG_PREFIX=$tool_path npm install --cache /tmp/empty-cache -g "${TOOL_NAME}@${TOOL_VERSION}"
```
Comment (viceice) [inline src/usr/local/buildpack/tools/yarn.sh:12]: We need to clean cache. Looks like it's not removed from install-tool, so next install command with different user fails.

Maybe better use a random path?

**Review by viceice — CHANGES_REQUESTED** [review]

File: src/usr/local/buildpack/tools/yarn.sh (hunk 1 above)
Comment (Chumper) [reply src/usr/local/buildpack/tools/yarn.sh]: Seems we need to clean this path `/home/user/.npm/_cacache`

Comment (Chumper) [reply src/usr/local/buildpack/tools/yarn.sh]: Seems we need to do a `npm cache clean --force` previously to the install

**Comment by Chumper** [comment]
@viceice I am not sure how reliable these tests are.
I would expect the user to be the same for a single image and not being changed during runtime.
In that case the test to switch to another user for openshift seems wrong to me.

**Comment by viceice** [comment]
The user will change on runtime on openshift.
Openshift will use a arbitrary user id to run the container.

But sure, it then won't run any command as our image user. Maybe we need to use `USER 1005:0` in tests to force root group?

**Comment by viceice** [comment]
I'm also unsure, why npm will access `~/.npm` 🤷‍♂️

File: src/usr/local/buildpack/tools/yarn.sh (hunk 1 above)
Comment (viceice) [reply src/usr/local/buildpack/tools/yarn.sh]: I thought when using the cache param it will use only that path, but npm seems to still use home dir.

Comment (viceice) [reply src/usr/local/buildpack/tools/yarn.sh]: Looks like a npm bug

- https://github.com/npm/cli/issues/3350

**Comment by Chumper** [comment]
Ok, let me just apply 775 for the folders

**Comment by viceice** [comment]
We need to be sure it now works for openshift.

**Comment by viceice** [comment]
Maybe add a new from line to docker tests and only use 1005 there, so we don't mix install method's

**Review by viceice — APPROVED** [review]

**Pull request merged**